lib/rules/no-constant-condition.js

```diff
@@ -31,8 +31,7 @@ module.exports = {
                 type: "object",
                 properties: {
                     checkLoops: {
-                        type: "boolean",
-                        default: true
+                        enum: ["all", "allExceptWhileTrue", "none"]
```

Comment:
it's a breaking change. can we allow `true`(same as "all"), `false`(same as "none") as suggested in the issue?

Reply:
sorry, i don't get it properly! are you asking for to set the values of `all` and `none` to `true` and `false` respectively? if this is the case then if we see following line from the issue:
> So we have agreed to extend the checkLoops option with three new values: all (like true), none (like false), and allExceptWhileTrue which will behave like all but will not flag while (true). allExceptWhileTrue will be the new default.

then seems like these values are correct.

Reply:
In previous versions, users could config:
```js
rules: { "no-constant-condition": ["error", { "checkLoops": false }] }
```

but it's no longer allowed in the change - it's a breaking change.

Reply:
i got this one
>can we allow true(same as "all"), false(same as "none") as suggested in the issue?

but confused in this line.

Reply:
Indeed we can't remove support for boolean values in `checkLoops` because we must maintain compatibility with existing configurations that use `true` or `false`. We don't want this change to be a breaking one.

The idea was to add support for three new new values but still allow `true` and `false` as synonyms for `"all"` and `"none"` respectively. Apologies if the confusion arose from my wording.

Reply:
Ah! so if i summarize it then `checkloops` can be configured in two ways:

option already has this one
```js
rules: { "no-constant-condition": ["error", { "checkLoops": true }] }
or
rules: { "no-constant-condition": ["error", { "checkLoops": false }] }
```
and also can be configured in this way
```js
rules: { "no-constant-condition": ["error", { "checkLoops": "all" }] }
or
rules: { "no-constant-condition": ["error", { "checkLoops": "allExceptWhileTrue" }] }
or
rules: { "no-constant-condition": ["error", { "checkLoops": "none" }] }
```
it is like using `oneOf` shchema

please correct if i am wrong.

Reply:
> Ah! so if i summarize it then `checkloops` can be configured in two ways:
> 
> option already has this one
> 
> ```js
> rules: { "no-constant-condition": ["error", { "checkLoops": true }] }
> or
> rules: { "no-constant-condition": ["error", { "checkLoops": false }] }
> ```
> 
> and also can be configured in this way
> 
> ```js
> rules: { "no-constant-condition": ["error", { "checkLoops": "all" }] }
> or
> rules: { "no-constant-condition": ["error", { "checkLoops": "allExceptWhileTrue" }] }
> or
> rules: { "no-constant-condition": ["error", { "checkLoops": "none" }] }
> ```
> 
> it is like using `oneOf` shchema

That's correct👍🏻

Reply:
I think it's right! 👍 
Thanks to @fasttime for helping me explain, as a non-native English speaker, forgive my bad English! 😅

Reply:
Done! added both ways to configure `checkLoops`.